frontend/apps/marketing/tests/e2e/scripts/component-tree/builder.ts

```diff
@@ -0,0 +1,91 @@
+import {Entry, Environment} from 'contentful-management';
+
+import {ALL_THE_THINGS_ENTRY_ID} from '../config';
+
+import {ComponentNode} from './component-tree';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function getLinkedEntryIdsFromExperience(datasources: any[]) {
+  const linkedEntryIdSet = new Set<string>();
+
+  for (const datasource of datasources) {
+    if (datasource.sys?.linkType === 'Entry') {
+      linkedEntryIdSet.add(datasource.sys.id);
+    }
+  }
+
+  return linkedEntryIdSet;
+}
+
+/**
+ * The Experience JSON contains a field called "dataSource" which
+ * is a map of datasources. Each datasource references another asset or entry in a
+ * tree structure. In this case, the "All The Things" experience entry is the root node
+ * and the datasources are the direct children.
+ */
```

Comment:
> In this case, the "All The Things" experience entry is the root node and the datasources are the direct children.

For my own understanding, are the datasources referenced here the files in `__snapshots__/entries.json`?

Reply:
The datasources are actually coming from the Contentful Experience entry for "All The Things" in its raw JSON.